src/main/kotlin/calculator/Calculator.kt

```diff
@@ -0,0 +1,66 @@
+package calculator
+
+import java.lang.IllegalArgumentException
+import java.math.BigDecimal
+import java.math.RoundingMode
+
+object Calculator {
+    private val operators: List<Operator> = listOf(
+        Operator.PlusOperator(),
+        Operator.MinusOperator(),
+        Operator.MultiplyOperator(),
+        Operator.DivideOperator()
+    )
+
+    fun calculate(argument: String): String {
+        val result = parse(argument).execute(::getOperator)
+
+        return format(result)
+    }
+
+    private fun format(value: BigDecimal): String {
+        val down = value.setScale(0, RoundingMode.DOWN)
+        return down.longValueExact().toString()
+    }
+
+    private fun parse(argument: String): CalculateExecutor {
+        val arguments = argument.trim().split(" ")
+
+        val operands = arguments.asSequence()
+            .filterIndexed { idx, _ -> idx % 2 == 0 }
+            .map { str -> BigDecimal.valueOf(str.toLong()) }
+            .toList()
+
+        val operators = arguments.asSequence()
+            .filterIndexed { idx, _ -> idx % 2 != 0 }
+            .toList()
+
+        return CalculateExecutor(operands, operators)
+    }
+
+    private fun getOperator(operator: String): Operator {
+        return operators.find { it.applicable(operator) }
+            ?: throw IllegalArgumentException("Unsupported Operator [$operator]")
+    }
+
+    private class CalculateExecutor(val operands: List<BigDecimal>, val operators: List<String>) {
+        init {
+            if (operands.isEmpty()) {
+                throw IllegalArgumentException("Must not be empty")
+            }
+            if (operands.size != operators.size + 1) {
+                throw IllegalArgumentException("Invalid Input")
+            }
+        }
+
+        fun execute(operandProvider: (input: String) -> Operator): BigDecimal {
+            var lastOperand = operands[0]
+
+            for (lastIndex in 0 until operators.size) {
```

Comment:
```suggestion
            for (lastIndex in operators.indices) {
```
인텔리제이가 이렇게도 쓸 수 있다고 추천해주네요 🧐